controller/query_blackbox_test.go

```diff
@@ -302,6 +303,201 @@ func (rest *TestQueryREST) TestShow() {
 	})
 }
 
+func (rest *TestQueryREST) TestUpdate() {
+
+	rest.T().Run("update query", func(t *testing.T) {
+		// 1 for each test, without conflict of changes during each test execution, isolated or not :)
+		testFxt := tf.NewTestFixture(t, rest.DB,
+			tf.CreateWorkItemEnvironment(),
+			tf.Queries(1))
+		svc, ctrl := rest.SecuredControllerWithIdentity(testFxt.Identities[0])
+		newTitle := "Query New 1001"
+		fields := `{"$AND": [{"space": "1b0efd64-ba69-42a6-b7da-762264744223"}]}`
+		payload := app.UpdateQueryPayload{
+			Data: &app.Query{
+				Attributes: &app.QueryAttributes{
+					Title:   newTitle,
+					Version: &testFxt.Queries[0].Version,
+					Fields:  fields,
+				},
+				ID:   &testFxt.Queries[0].ID,
+				Type: query.APIStringTypeQuery,
+			},
+		}
+		resp, updated := test.UpdateQueryOK(t, svc.Context, svc, ctrl, testFxt.Spaces[0].ID, testFxt.Queries[0].ID, nil, nil, &payload)
+		assert.Equal(t, newTitle, updated.Data.Attributes.Title)
+		compareWithGoldenAgnostic(t, filepath.Join(rest.testDir, "update", "ok.query.golden.json"), updated)
```

Comment:
Please use this ending pattern `your_test_name.<res|req>.<payload|headers>.golden.json`.